pkg/yang/entry.go

```diff
@@ -1024,11 +1026,30 @@ func addExtraKeywordsToLeafEntry(n Node, e *Entry) {
 			"reference",
 			"status",
 			"when":
-			e.Extra[name] = append(e.Extra[name], fv.Interface())
+			if !fv.IsNil() {
+				addToExtrasSlice(fv, name, e)
+			}
 		}
 	}
 }
 
+func addToExtrasSlice(fv reflect.Value, name string, e *Entry) {
+	if fv.Kind() == reflect.Slice {
+	outerLoop:
+		for j := 0; j < fv.Len(); j++ {
+			for _, en := range e.Extra[name] {
+				if en == fv.Index(j).Interface() {
+					// Don't add again if identical
+					continue outerLoop
+				}
```

Comment:
What's the rationale for removing duplicates? Maybe a particular extension has a different meaning when duplicated?

Reply:
When running the tests it became clear that the existing implementation allowed a 2nd entry to overwrite the first. It was not duplicating, just overwriting.

Reply:
That's really curious. I'm realizing now that this `==` comparison is most likely comparing two slices, of a pointer value, which sticks out to me as something that may not be what's intended. How could the existing implementation, or now your loop implementation, overwrite when it's doing an append?

Reply:
@wenovus Thanks for replying so again so soon.
I found that the duplicate entry happened in `TestIfFeature()` when testing `LeafList`.
`LeafList` calls `ToEntry()` directly which calls `addExtraKeywordsToLeafEntry()` and then calls `addExtraKeywordsToLeafEntry()` again with the same "ifFeature" statment.

With the code on `master` when it populates `Extra` it adds every entry twice - then very cleverly the test case (`entry_test.go:2009`) only takes the first entry and discards the 2nd.

When I changed the way entries are written to the `Extra`, I changed the test case so it does not ignore any entries, but then I found that I was getting: 
```
entry_test.go:2146: leaf-list: did not get expected if-features, got [ft-leaf-list ft-leaf-list], want [ft-leaf-list]

```
because `LeafList` was adding the `Extra` twice.
That's why I added the statement to "Don't add again if identical" - yes it's comparing pointers - because that's what I wanted not to have the exact same thing added twice.

Reply:
@wenovus both have been removed now and all tests are passing